config/confighttp/confighttp.go

```diff
@@ -32,6 +33,28 @@ import (
 	"go.opentelemetry.io/collector/internal/middleware"
 )
 
+type CompressionType string
+
+const (
+	CompressionUnsupported CompressionType = ""
```

Comment:
```suggestion
	CompressionUnsupported CompressionType = "<unsupported>"
```

I would use a value here that is unlikely to appear accidentally.

Reply:
deleted the usage of `CompressionUnsupported`